### ISSUE TITLE
Custom encodings not working

### ISSUE THREAD

**Issue by burgholzer** [cda-tum/qmap#58]
There seems to be a bug in the custom `commander` and `bimander` encodings. The following MWE (added as an additional test case to the `test_exact.cpp` file) produces very unexpected results:

```c++
TEST_F(ExactTest, MWE) {
    using namespace dd::literals;

    auto circ = qc::QuantumComputation(3U);
    circ.x(2, 1_pc);
    circ.x(1, 0_pc);

    auto arch = Architecture("../../extern/architectures/ibmq_casablanca.arch");

    auto mapper = ExactMapper(circ, arch);
    auto settings = Configuration();
    settings.useSubsets = false;
    settings.encoding = Encoding::Commander; // or Encoding::Bimander
    settings.commanderGrouping = CommanderGrouping::Halves; // or CommanderGrouping::Fixed2, CommanderGrouping::Fixed3, CommanderGrouping::Logarithm
    mapper.map(settings);
    mapper.printResult(std::cout);
}
```

The results (running under `macOS 12.3.1` using `z3-4.8.17`) can be summarized as follows:

Encoding | Grouping | Result
--- | --- | ---
None | - | Optimal solution (0 SWAPs) within 2.5s
Commander | Halves | Suboptimal solution (1 SWAP) within 2.5s
\- | Fixed2 | z3 returns `UNSAT` within a fraction of a second
\- | Fixed3 | Optimal solution (0 SWAPs) within 3.2s
\- | Logarithm | Takes forever (>5min) then segfaults
Bimander | Halves | Suboptimal solution (2 SWAPs) within 2.3s
\- | Fixed2 | z3 returns `UNSAT` within a fraction of a second
\- | Fixed3 | Suboptimal solution (2 SWAPs) within 2.0s
\- | Logarithm | Takes forever (>5min) then segfaults

**Comment by burgholzer** [comment]
This is fixed by #59.

**Comment by burgholzer** [comment]
Unfortunately, the underlying issue is still not fixed. Mapping the following circuit produces `UNSAT` for all encoding variants:
```c++
    qc = qc::QuantumComputation(5U);
    qc.x(1, 0_pc);
    qc.x(2, 0_pc);
    qc.x(3, 0_pc);
    qc.x(4, 0_pc);

    arch = Architecture("../../extern/architectures/ibmq_casablanca.arch");

    mapper = std::make_unique<ExactMapper>(qc, arch);

    settings.verbose = true;
    settings.method  = Method::Exact;
    settings.useSubsets = false;
```
I identified the problem and a fix is on the way.
